test/core/bearerTokenHandler/test.js

```diff
@@ -0,0 +1,330 @@
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2019 (original work) Open Assessment Technologies SA ;
+ */
+
+/**
+ * @author Tamas Besenyei <tamas@taotesting.com>
+ */
+
+define(['jquery', 'core/bearerTokenHandler', 'jquery.mockjax'], ($, bearerTokenHandlerFactory) => {
+    'use strict';
+
+    QUnit.module('factory');
+
+    QUnit.test('module', assert => {
+        assert.expect(1);
+        assert.ok(typeof bearerTokenHandlerFactory === 'function', 'the module exposes a function');
+    });
+
+    QUnit.test('instantiate', assert => {
+        assert.expect(2);
+        assert.ok(typeof bearerTokenHandlerFactory() === 'object', 'the factory produces an object');
+        assert.notEqual(
+            bearerTokenHandlerFactory(),
+            bearerTokenHandlerFactory(),
+            'the factory produces a different object at each call'
+        );
+    });
+
+    // prevent the AJAX mocks to pollute the logs
+    $.mockjaxSettings.logger = null;
+    $.mockjaxSettings.responseTime = 1;
+
+    QUnit.module('API', {
+        beforeEach: function() {
+            this.handler = bearerTokenHandlerFactory({ refreshTokenUrl: '//refreshUrl' });
+        },
+        afterEach: function() {
+            this.handler.clearStore();
```

Comment:
I guess the async pattern applies here, like you did in the bearerTokenStore test:
```
afterEach: function(assert) {
    const done = assert.async();
    this.handler.clearStore().then(done);
    $.mockjax.clear();
}
```